### DISCUSSION TITLE
How to get image file names from DataLoader

### DISCUSSION THREAD

**Discussion by karllandheer** [Project-MONAI/MONAI#7878] [General]
Hello, I would like to get the image filenames from the DataLoader. What's the easiest way to do this? Is it safe to assume the order of the images is the same as input when accessing the data by iterating through the data loader object as long as shuffle = False like this. I.e., will all_labels and local_train_images match in ordering?

```
train_ds = ImageDataset(image_files=local_train_images, labels=train_labels, transform=train_transforms) 
train_loader = DataLoader(train_ds, batch_size=8, shuffle=False, num_workers=2, pin_memory=pin_memory) 

all_labels = []
for batch_data in train_loader:
            step += 1
            inputs, labels = batch_data[0].to(device), batch_data[1].to(device)
            all_labels.append(labels)
```

**Comment by KumoLiu** [comment]
Hi @karllandheer, you can get the file name by using `inputs.meta["filename_or_obj"]`. The info should be saved in the meta information.

Hope it helps, move to discussion for now.
Feel free to create another one if you have more issues.
Thanks.
